bundles/AdminBundle/Controller/Admin/DataObject/ClassificationstoreController.php

```diff
@@ -1023,7 +1024,12 @@ public function addCollectionsAction(Request $request)
                     $context['keyId'] = $keyData->getKeyId();
                     $context['groupId'] = $groupId;
                     $context['keyDefinition'] = $definition;
-                    if (method_exists($definition, 'enrichLayoutDefinition')) {
+                    //TODO Pimcore 11: remove method_exists BC layer
+                    if ($definition instanceof DataObject\ClassDefinition\Data\LayoutDefinitionEnrichmentInterface || method_exists($definition, 'enrichLayoutDefinition')) {
+                        if (!$definition instanceof DataObject\ClassDefinition\Data\LayoutDefinitionEnrichmentInterface) {
+                            trigger_deprecation('pimcore/pimcore', '10.1', sprintf('Usage of method_exists is deprecated since version 10.1 and will be removed in Pimcore 11.' .
+                                'Implement the %s interface instead.', DataObject\ClassDefinition\Data\LayoutDefinitionEnrichmentInterface::class));
```

Comment:
`trigger_deprecation()` uses `printf` internally, just pass additional arguments after the message:

```suggestion
                            trigger_deprecation('pimcore/pimcore', '10.1', 'Usage of method_exists is deprecated since version 10.1 and will be removed in Pimcore 11.' .
                                'Implement the %s interface instead.', DataObject\ClassDefinition\Data\LayoutDefinitionEnrichmentInterface::class);
```

Reply:
This is also true for all other calls to `trigger_deprecation` ;)

Reply:
true, let's create a follow up a follow up ticket. not even sure if it makes sense to invest time on deprecation notices style.